drivers/soundwire/intel_auxdevice.c

```diff
@@ -109,12 +112,19 @@ static int sdw_master_read_intel_prop(struct sdw_bus *bus)
 		return -EIO;
 	}
 
-	fwnode_property_read_u32(link,
-				 "intel-sdw-ip-clock",
-				 &prop->mclk_freq);
+	if (!sdw->link_res->cardinal_clock_capable) {
+		fwnode_property_read_u32(link,
+					 "intel-sdw-ip-clock",
+					 &prop->mclk_freq);
 
-	/* the values reported by BIOS are the 2x clock, not the bus clock */
-	prop->mclk_freq /= 2;
+		/* the values reported by BIOS are the 2x clock, not the bus clock */
+		prop->mclk_freq /= 2;
+	} else {
+		prop->mclk_freq = 24576000;
+		prop->clk_freq[0] = 6144000;
```

Comment:
@plbossart We need to set max_clk_freq as well. Because `sdw_master_read_prop` set max_clk_freq before here.
```patch
-               prop->clk_freq[0] = 6144000;
+               prop->max_clk_freq = prop->clk_freq[0] = 6144000;
```
However, I still got timeout error after applying the change.
```
[  161.438494] soundwire sdw-master-0: Controller Timed out on bank switch
[  161.438525] soundwire sdw-master-0: multi link bank switch failed: -110
[  161.438547] _sdw_prepare_stream: do_bank_switch failed: -110
```

Reply:
Humm, yes. I'll recheck what properties we need. Maybe it's just better to just override the values in mipi-disco.c initially to debug all this.

Reply:
@bardliao I managed to get something working with the mclk_freq set to 12288000. All tests pass in my remote tests.

We'll need to ask @fredoh9 to measure the clock to make sure the output is indeed 6.144 MHz. I am not sure if the SYNCPRD is exact though, it could be off by a factor of two. Not that it really matters, it would change the frequency to either 2 or 8kHz, no one would see the difference.

Reply:
For internal RT711 soundwire codec, AFAIK, there is no hw probe point unless we ask for rework. Let me check/ask that to hw team and find a way

Reply:
@fredoh9 this setting would be applied to the 3-in-1 board as well, not sure if we have any connected to MTL?